pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -262,14 +262,6 @@ func getInt64(i int64) *int64 {
 var (
 	crv = apiextensionsv1beta1.CustomResourceValidation{
 		OpenAPIV3Schema: &apiextensionsv1beta1.JSONSchemaProps{
-			OneOf: []apiextensionsv1beta1.JSONSchemaProps{
```

Comment:
@raybejjani Please fix this in the endpoint CRD PR as well

Reply:
Sure. I don't get this, though. Is it because they'll begin enforcing the presence of a spec/specs field? In the endpoint CRD case we only have one field anyway (and so it's anyway not quite correct to use the OneOf) but it is status. Is there a link explaining what's allowed and isn't?
Separately, do we do anything special for the <1.10 versions to retain the old behaviour?

Reply:
@aanm Putting a blocking review until the above is clarified

Reply:
@tgraf is in the commit message. Should I point to the k8s PR that states that?

Reply:
Yes. Do we need to preserve the existing code for older k8s versions?

Reply:
aanm should answer also, but my reading of it is that, at worst, we would loosen the validation on older versions. This would only be around requiring `spec` or `specs` to exist. The actual content validation would still apply. It's not ideal but it's unlikely someone would just not include a spec field since that's the whole point.

Reply:
@tgraf does the explanation given by @raybejjani makes sense?

Reply:
I'm OK with it.